smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/json/MemberShapeDecodeGenerator.kt

```diff
@@ -179,11 +179,21 @@ abstract class MemberShapeDecodeGenerator(
                     } else { // decode date as a string manually
                         val dateName = "date$level"
                         val swiftTimestampName = TimestampHelpers.generateTimestampFormatEnumValue(timestampFormat)
-                        writer.write(
-                            "let \$L = try containerValues.timestampStringAsDate(\$L, format: .\$L, forKey: .\$L)",
-                            dateName, iteratorName, swiftTimestampName, topLevelMember.memberName
-                        )
-                        writer.write("${decodedMemberName}$terminator.$insertMethod($dateName)")
+                        if (!isSparse) {
```

Comment:
Timestamps were not being decoded properly when the source string comes from a sparse array.  In that case, the source is now properly unwrapped before conversion to a Swift `Date`.